### ISSUE TITLE
why the detection result conf is unstable

### ISSUE THREAD

**Issue by jaffe-fly** [ultralytics/ultralytics#19890]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

my model is `yolo11l-OBB`
when detect object from a video frame , found the result conf has a big change,form 0.93 down to 0.75, even if the object not move,  

![Image](https://github.com/user-attachments/assets/ee472217-e5e3-499f-9749-5827b7663c16)

sometimes, the conf change  will  lower than the set threshold

I would like to know, during training, which parameters can be adjusted to reduce such variations and enhance the robustness of the model

### Additional
now my training set is below, `Albumentations` use default set
```
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, num_output_channels=3, method='weighted_average'), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
```
```
args = dict(data=data,
                project=project,
                # exist_ok=True,
                epochs=400,
                device=3,
                verbose=True,
                workers=8,
                cache='disk',
                amp=True,
                batch=12,
                plots=True,
                imgsz=640,
                degrees=180,
                translate=0,
                mosaic=0,
                mixup=0,
                scale=0.5,
                multi_scale=True,
                shear=0,
                flipud=0.1,
                fliplr=0.1,
                bgr=0.1,
                optimizer='auto', # SGD, Adam, AdamW, NAdam, RAdam, RMSProp etc.
                augment=True,
                # warmup_epochs=10,
                # cos_lr=True,
                agnostic_nms=True,
                # close_mosaic=20,
                )
```

**Comment by UltralyticsAssistant** [comment]
👋 Hello @jaffe-fly, thank you for your interest in Ultralytics 🚀! We recommend checking out the [Docs](https://docs.ultralytics.com/) for detailed guidance, including information on [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions and scenarios are already addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further. This example should include details like your model configuration, dataset, and any custom training commands used.

For custom training ❓ Questions like yours, please include as much relevant detail as possible. This could include example frames from your dataset, training logs, and the specific hyperparameters you've experimented with. We also recommend reviewing our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal performance.

Join the Ultralytics community to connect with other users and experts! For real-time chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧. For more in-depth discussions, check out [Discourse](https://community.ultralytics.com/), or explore threads on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Ensure you're using the latest version of the `ultralytics` package, which may already address your issue. Update it in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run seamlessly in the following verified environments, complete with preinstalled dependencies ([CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/)):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests validate functionality across all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to assist you quickly. An Ultralytics engineer will review your issue and provide further guidance soon. Stay tuned! 😊

**Comment by Y-T-G** [comment]
It's probably due to overfitting. You need more augmentations

**Comment by jaffe-fly** [comment]
> [@jaffe-fly](https://github.com/jaffe-fly) thank you for providing detailed training configuration. For more stable confidence scores in YOLO11l-OBB, consider these adjustments:
> 
> 1. **Increase Augmentation Diversity**: Enable mosaic (`mosaic=1.0`) and mixup (`mixup=0.1`) to improve generalization
> 2. **Optimizer Tuning**: Switch to `optimizer='AdamW'` with reduced `lr0=0.001` for smoother convergence
> 3. **Learning Rate Schedule**: Uncomment `cos_lr=True` to apply cosine annealing
> 4. **Loss Weight Adjustment**: Increase classification weight (`cls=1.0`) relative to box loss
> 5. **Stabilization**: Add `close_mosaic=10` to disable mosaic in final epochs
> 
> These changes help balance detection consistency and accuracy. For detailed hyperparameter guidance, see the Ultralytics Hyperparameter Tuning Guide and Model Evaluation Insights documentation.

According to my tests, the issue mentioned above still persists
next, i will test some augments in `albumentations`

**Comment by jaffe-fly** [comment]
```
transform = A.SomeOf([
        A.RandomBrightnessContrast(p=1),
        A.MotionBlur(p=1),
        A.GaussianBlur(p=1),
        A.ISONoise(p=1),
        A.RandomFog(p=1),
        A.RandomGamma(p=1),
        A.ImageCompression(quality_lower=75, p=1),
    ], p=0.2,n=3)
```
can modify code `ultralytics\data\augment.py` L1863

**Comment by glenn-jocher** [comment]
@jaffe-fly thank you for sharing your augmentation modifications. Yes, you can modify line 1863 in `augment.py` to implement your custom Albumentations pipeline. Your proposed implementation using `A.SomeOf` with multiple transformations is a valid approach to enhance augmentation diversity. For optimal results, ensure the transformation probabilities and combinations align with your dataset characteristics. 

For guidance on integrating custom Albumentations transforms with YOLO11, see the Ultralytics Albumentations Integration documentation. This resource explains how to properly structure transformations while maintaining compatibility with YOLO's data pipeline. 

Would you like me to clarify any specific implementation details for your modified augmentation chain?

**Comment by jaffe-fly** [comment]
> [@jaffe-fly](https://github.com/jaffe-fly) thank you for sharing your augmentation modifications. Yes, you can modify line 1863 in `augment.py` to implement your custom Albumentations pipeline. Your proposed implementation using `A.SomeOf` with multiple transformations is a valid approach to enhance augmentation diversity. For optimal results, ensure the transformation probabilities and combinations align with your dataset characteristics.
> 
> For guidance on integrating custom Albumentations transforms with YOLO11, see the Ultralytics Albumentations Integration documentation. This resource explains how to properly structure transformations while maintaining compatibility with YOLO's data pipeline.
> 
> Would you like me to clarify any specific implementation details for your modified augmentation chain?

Yes, Thank you very much

**Comment by glenn-jocher** [comment]
You're welcome! For your custom Albumentations pipeline, ensure transformations maintain YOLO's bounding box format by using `A.BboxParams(format='yolo')` when spatial transforms are present. The Ultralytics Albumentations Integration documentation shows specific examples of implementing `A.SomeOf` while preserving label compatibility. Test incrementally and let me know if you encounter specific compatibility issues during implementation.
